pkg/backends/influxdb/flux/parser.go

```diff
@@ -0,0 +1,123 @@
+package flux
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/trickstercache/trickster/v2/pkg/timeseries"
+	"github.com/trickstercache/trickster/v2/pkg/util/timeconv/duration"
+)
+
+type Parser struct {
+	reader io.Reader
+}
+
+func NewParser(reader io.Reader) *Parser {
+	return &Parser{
+		reader: reader,
+	}
+}
+
+func (p *Parser) ParseQuery() (*Query, error) {
+	valid := false
+	r := bufio.NewReader(p.reader)
+	ln := 0
+	q := &Query{}
+	for line, err := r.ReadString('\n'); ; line, err = r.ReadString('\n') {
```

Comment:
since a flux query is generally very small (almost always under 1k), I'm not sure if it's necessary to iterate over the query lines here. To avoid memory churn and unnecessary checks for the newline character as an interruptor, I think we can just io.ReadAll() and apply the subsequent string.Contains() checks to the whole query string. That would be a bit more performant at scale.